server/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java

```diff
@@ -2364,4 +2366,28 @@ public void test_select_partitioned_by_column_with_query_then_fetch_plan() throw
             "1| Trillian\n"
         ));
     }
+
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_alter_table_drop_constraint_on_partitioned_table() throws Exception {
+        // Dropping a constraint on a partitioned table used to fail before 5.1
+        execute("create table t (id int primary key constraint check_id_gt_zero check(id > 0), " +
+            "x int constraint check_x_gt_zero check(x > 0)) " +
+            "partitioned by (id) " +
+            "clustered into 1 shards " +
+            "with (number_of_replicas=0)");
+        refresh();
+
+        String selectCheckConstraintStmt =
+            "select table_schema, table_name, constraint_type, constraint_name " +
+                "from information_schema.table_constraints " +
+                "where table_name='t' and constraint_name = 'check_x_gt_zero' " +
+                "order by constraint_name";
+
+        execute("alter table t drop constraint check_x_gt_zero");
+        refresh();
```

Comment:
```suggestion
        String selectCheckConstraintStmt =
            "select table_schema, table_name, constraint_type, constraint_name " +
                "from information_schema.table_constraints " +
                "where table_name='t' and constraint_name = 'check_x_gt_zero' " +
                "order by constraint_name";

        execute("alter table t drop constraint check_x_gt_zero");
```

`refresh` is refreshing the segments/readers, that shouldn't be necessary to observe schema changes.